examples/audio/eeg_signal.py

```diff
@@ -0,0 +1,554 @@
+We train a model from scratch since such signal-classification models are fairly scarce
+in pre-trained format.
+The data we use is sourced from the UC Berkeley-Biosense Lab where the data was collected
+from 15 subjects at the same time.
+Our process is as follows:
+
+- Load the [UC Berkeley-Biosense Synchronized Brainwave
+Dataset](https://www.kaggle.com/datasets/berkeley-biosense/synchronized-brainwave-dataset)
+Dataset](https://www.kaggle.com/datasets/berkeley-biosense/synchronized-brainwave-dataset)
+- Visualize random samples from the data
+- Pre-process, collate and scale the data to finally make a `tf.data.Dataset`
+- Prepare class weights in order to tackle major imbalances
+- Create a Conv1D and Dense-based model to perform classification
+- Define callbacks and hyperparameters
+- Train the model
+- Plot metrics from History and perform evaluation
+
+This example needs the following external dependencies (Gdown, Scikit-learn, Pandas,
+Numpy, Matplotlib). You can install it via the following commands.
+
+Gdown is an external package used to download large files from Google Drive. To know
+more, you can refer to its [PyPi page here](https://pypi.org/project/gdown)
+"""
+
+"""
+## Setup and Import headers
+"""
+
+"""shell
+pip install gdown
+pip install gdown
+pip install sklearn
+pip install pandas
+pip install numpy
+pip install matplotlib
+gdown 1V5B7Bt6aJm0UHbR7cRKBEK8jx7lYPVuX
+# gdown will download eeg-data.csv onto the local drive for use. Total size of
+eeg-data.csv is 105.7 MB
+"""
+
+import pandas as pd
+import matplotlib.pyplot as plt
+import json
+import numpy as np
+from tensorflow import keras
+from tensorflow.keras import layers
+import tensorflow as tf
+from sklearn import preprocessing, model_selection
+import random
+
+QUALITY_THRESHOLD = 128
+BATCH_SIZE = 64
+SHUFFLE_BUFFER_SIZE = 100
+
+"""
+## Read data from `eeg-data.csv`
+
+We use the Pandas library to read the `eeg-data.csv` file and display the first 5 rows
+using the `.head()` command
+"""
+
+eeg = pd.read_csv("eeg-data.csv")
+
+"""
+We remove unlabeled samples from our dataset as they do not contribute to the model. We
+also perform a `.drop()` operation on the columns that are not required for training data
+preparation
+"""
+
+unlabeled_eeg = eeg[eeg["label"] == "unlabeled"]
+eeg = eeg.loc[eeg["label"] != "unlabeled"]
+eeg = eeg.loc[eeg["label"] != "everyone paired"]
+
+eeg.drop(
+    [
+        "indra_time",
+        "Unnamed: 0",
+        "browser_latency",
+        "reading_time",
+        "attention_esense",
+        "meditation_esense",
+        "updatedAt",
+        "createdAt",
+    ],
+    axis=1,
+    inplace=True,
+)
+
+eeg.reset_index(drop=True, inplace=True)
+eeg.head()
+
+"""
+In the data, the samples recorded are given a score from 0 to 128 based on how
+well-calibrated the sensor was (0 being best, 200 being worst). We filter the values
+based on an arbitrary cutoff limit of 128.
+"""
+
+
+def makeNumpy(value_string):
```

Comment:
lets give this a more descriptive name.

also, remember to snake case!

Reply:
Will do!